src/layout/layout.js

```diff
@@ -99,44 +98,47 @@ function _layout() {
     return layout;
 }
 
-function layoutAdapter(name, value) {
-    // Quick bodge to fix #568
-    if (this.node() == null) {
-        return this;
+function layoutSelection(name, value) {
+    var argsLength = arguments.length;
+
+    // For layout(string), return the lyout value for the first node
```

Comment:
Edge case but would this blow up if you do `x.enter().append('g').layout('width')`?